### PR TITLE
specifyTargetComponent add-on

### PR DESCRIPTION
**Pull request by opotowsky**
## Description

MiniPR to follow PR #677: 
This updates the tests in the `TestSpecifyTargetComponent` class to be more clear about the intended outcome of each test. After user-developer consultation with @albeanth, also adding the `SLUG` flag to the list of preferred target flags. 

---

## Checklist

- [x] The code is understandable and maintainable to people beyond the author.
- [x] Tests have been added/updated to verify that the new or changed code works.
- [x] There is no commented out code in this PR.
- [x] The commit message follows [good practices](https://terrapower.github.io/armi/developer/tooling.html).
- [x] All docstrings are still up-to-date with these changes.

If user exposed functionality was added/changed:

- [ ] Documentation added/updated in the `doc` folder.
- [ ] New or updated dependencies have been added to `setup.py`.

### CODE REVIEW

File: armi/reactor/converters/tests/test_axialExpansionChanger.py [hunk 1]
```diff
@@ -567,14 +567,15 @@ class TestSpecifyTargetComponent(unittest.TestCase):
...
 
     def setUp(self):
-        Base.setUp(self)
+        self.obj = AxialExpansionChanger()
         self.a = buildTestAssemblyWithFakeMaterial(name="FakeMatException")
         self.obj.setAssembly(self.a)
+        # need an empty dictionary because we want to test for the added component only
+        self.obj.expansionData._componentDeterminesBlockHeight = (
+            {}
+        )  # pylint: disable=protected-access
```
Comment (opotowsky) [inline armi/reactor/converters/tests/test_axialExpansionChanger.py:576]: @john-science is the pylint comment necessary here? @albeanth and I aren't sure

**Comment by albeanth** [comment]
@john-science I can't officially review, but I approve this PR (if that means anything 😄)

File: armi/reactor/converters/tests/test_axialExpansionChanger.py (hunk 1 above)
Comment (john-science) [reply armi/reactor/converters/tests/test_axialExpansionChanger.py:576]: I am not particularly bothered with `pylint` concerns in unit tests, so I tend to put one big `pylint:disable` at the top of each file:

https://github.com/terrapower/armi/blob/e0dd171ff4fa1433f59e15420a8dcc7428b909af/armi/reactor/converters/tests/test_geometryConverters.py#L16

Seems easier.

**Review by john-science — APPROVED** [review]
Approved.

Though, sure, you could clean up that pylint a bit. See my comment.

**Pull request merged**